cli/azd/pkg/tools/git/git.go

```diff
@@ -143,6 +146,29 @@ func (cli *gitCli) InitRepo(ctx context.Context, repositoryPath string) error {
 	return nil
 }
 
+func (cli *gitCli) CheckConfigCredentialHelper(ctx context.Context) (bool, error) {
+	found, err := tools.ToolInPath("git")
+	if !found {
+		return false, err
+	}
+	gitRes, err := tools.ExecuteCommand(ctx, "git", "config", "credential.helper")
+	if err != nil {
+		return false, fmt.Errorf("checking %s credential.helper: %w", cli.Name(), err)
+	}
+
+	return gitRes != "", nil
+}
+
+func (cli *gitCli) SetCredentialStore(ctx context.Context, repositoryPath string) error {
+	runArgs := exec.NewRunArgs("git", "-C", repositoryPath, "config", "credential.helper", "store")
```

Comment:
Curious... How did we land on `git-credential-store`? I'm unfamiliar with the tool but does it by default simply just store the credentials as plaintext on the filesystem? https://git-scm.com/docs/git-credential-store Just trying to understand the usage here. Also, does it also affect the normal git credentials (outside of pipeline)?

Reply:
This is helping to prevent asking for git credentials multiple times.
The azdo PAT should be used as git-credential and set to be default for the repo with git-credential-store.

Reply:
1. How is this stored on the filesystem?
2. Does this also affect not only PAT, but the username / password if the user uses username / password auth for their git repository?

Reply:
@vhvb1989 do you have context on what this might be used for?

Reply:
It is a way to persist the initial PAT as the git credentials for the remote, so user is not required for password again.